sdk/python/kfp/v2/google/aiplatform.py

```diff
@@ -0,0 +1,464 @@
+    output_parameters: The output parameter declaration. Should be a mapping
+      from output name to one of 1) str, 2) float, or 3) int.
+    image_uri: The URI of the container image containing the user training
+      program. Applicable for custom container training.
+    commands: The container command/entrypoint. Applicable for custom container
+      training.
+    executor_image_uri: The URI of the container image containing the
+      dependencies of user training program. Applicable for custom Python
+      training.
+    package_uris: The Python packages that are expected to be running on the
+      executor container. Applicable for custom Python training.
+    python_module: The entrypoint of user training program. Applicable for
+      custom Python training.
+    args: The command line arguments of user training program. This is expected
+      to be a list of either 1) constant string, or 2) KFP DSL placeholders, to
+      connect the user program with the declared component I/O.
+    machine_type: The machine type used to run the training program. The value
+      of this field will be propagated to all worker pools if not specified
+      otherwise in additional_job_spec.
+    additional_job_spec: Full-fledged custom job API spec. The value specified
+      in this field will override the defaults provided through other function
+      parameters.
+
+      For details please see:
+      https://cloud.google.com/ai-platform-unified/docs/reference/rest/v1beta1/CustomJobSpec
+
+  Returns:
+    A KFP ContainerOp object represents the launcher container job, from which
+    the user training program will be submitted to AI Platform (Unified) Custom
+    Job service.
+
+  Raises:
+    KeyError on name collision between parameter and artifact I/O declaration.
+    ValueError when:
+      1. neither or both image_uri and executor_image_uri are provided; or
+      2. no valid package_uris and python_module is provided for custom Python
+         training.
+  """
+  # Check the sanity of the provided parameters.
+  input_artifacts = input_artifacts or {}
+  input_parameters = input_parameters or {}
+  output_artifacts = output_artifacts or {}
+  output_parameters = output_parameters or {}
+  if bool(set(input_artifacts.keys()) & set(input_parameters.keys())):
+    raise KeyError('Input key conflict between input parameters and artifacts.')
+  if bool(set(output_artifacts.keys()) & set(output_parameters.keys())):
+    raise KeyError('Output key conflict between output parameters and '
+                   'artifacts.')
+
+  if not additional_job_spec and  bool(image_uri) == bool(executor_image_uri):
+    raise ValueError('The user program needs to be either a custom container '
+                     'training job, or a custom Python training job')
+
+  # For Python custom training job, package URIs and modules are also required.
+  if executor_image_uri:
+    if not package_uris or not python_module or len(package_uris) > 100:
```

Comment:
nit: make a constant variable for the magic number: 100.

Reply:
Done.